app/src/main/java/com/hedvig/app/feature/birthday/viewmodel/BirthdayViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.hedvig.app.feature.birthday.viewmodel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.hedvig.app.feature.birthday.data.BirthdayRepository
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.plusAssign
+import timber.log.Timber
+import java.util.*
+
+
+class BirthdayViewModel(
+    private val birthdayRepository: BirthdayRepository
+) : ViewModel() {
+
+    private lateinit var birthDate: String
+
+    val isBirthdayData: MutableLiveData<Boolean> = MutableLiveData()
+    private val disposables = CompositeDisposable()
+
+
+
+    private fun isBirthday(): Boolean {
+
+        val birthMonth = birthDate.substring(4, birthDate.length - 6).toInt()
+        val birthDay = birthDate.substring(6, birthDate.length - 4).toInt()
+
+        val calendar = Calendar.getInstance()
+
+        val currentMonth = calendar.get(Calendar.MONTH) + 1
+        val currentDay = calendar.get(Calendar.DAY_OF_MONTH)
+
+        return currentDay == birthDay && currentMonth == birthMonth
+    }
+
+    fun getBirthDay() {
```

Comment:
This function only appears to be called once when the ViewModel is created. Since that is the case, we can move the function call to an `init`-block in the ViewModel and remove the explicit call in the `SplashActivity`.